lib/solveEquation/stepThrough.js

```diff
@@ -50,9 +51,23 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
   const MAX_STEP_COUNT = 20;
   let iters = 0;
 
+  // Checks if there are roots in the original equation before we
```

Comment:
This code is repeated here and below, why? Can we break it out in a helper function?